cmd/thanos/query.go

```diff
@@ -891,14 +906,12 @@ func prepareEndpointSet(
 			}
 
 			for _, eg := range endpointGroupAddrs {
-				addr := fmt.Sprintf("dns:///%s", eg)
```

Comment:
Note: this is a breaking change; its so that people can chose to keep previous behavior by adding "dns:///" prefix or opt into thanos resolver by adding "thanos:///" prefix; alternatively we could snoop if thanos:/// is added and if not add dns:/// or just add "thanos:///" by default, in theory it should be not noticeable mostly. I like it better to state the resolver explicitely here.

Reply:
What's the main difference between the Thanos DNS resolver and the default DNS resolver we are using?

Trying to understand if it is a big difference. If it is, then I think the detection is needed and if no prefix is added we can still default to `dns://`. If there is no big difference I think I am fine either way

Reply:
The thanos one understands our "dnssrv+" stuff while the dns one does not. Otherwise it should be roughly the same.